File: Java/MysticDave/Animation.java
/*
========================================================================

	A sequence of sprites to play through
	Each frame has a set duration

========================================================================
*/
package MysticDave;

public class Animation {
    
    public int numFrames;
    
    public char[]       frameIDs;
    public TextureSheet tex;
    public int[]        frameDurations;
    public int          totalDuration;    
    
    public Animation( int numFrames, TextureSheet tex ) {
        this.numFrames = numFrames;
        this.tex       = tex;
        this.frameDurations = new int[numFrames];
        this.frameIDs  = new char[numFrames];
    }
    
    public void Init() {
        totalDuration = 0;
        for ( int i = 0; i < numFrames; i ++ ) {
            totalDuration += frameDurations[i];
        }
    }
    
}

File: CPP/Animation.cpp
/*
========================================================================

	A sequence of sprites to play through
	Each frame has a set duration

    Author: Tim Wheeler
    Contact: timwheeleronline@gmail.com

========================================================================
*/

#include "Animation.h"

using namespace MysticDave;

Animation::Animation( int numFrames, TextureSheet * tex ) {
	Animation::numFrames = numFrames;
	Animation::tex = tex;
	Animation::frameDurations = new int[numFrames];
	Animation::frameIDs = new char[numFrames];
}

Animation::~Animation() {
	delete[] frameDurations;
	delete[] frameIDs;
}

void Animation::Init() {
	totalDuration = 0;
	for ( int  i = 0;  i < numFrames; i ++ ) {
		totalDuration += frameDurations[i];
	}
}